pytorch_lightning/loggers/csv.py

```diff
@@ -0,0 +1,204 @@
+"""
+CSV logger
+----------
+
+CSV logger for basic experiment logging that does not require opening ports
+
+"""
+import io
+import os
+import csv
+import torch
+
+from argparse import Namespace
+from typing import Optional, Dict, Any, Union
+
+
+from pytorch_lightning import _logger as log
+from pytorch_lightning.core.saving import save_hparams_to_yaml
+from pytorch_lightning.loggers.base import LightningLoggerBase
+from pytorch_lightning.utilities.distributed import rank_zero_warn, rank_zero_only
+
+
+class ExperimentWriter(object):
+    r"""
+    Experiment writer for CSVLogger.
+
+    Currently supports to log hyperparameters and metrics in YAML and CSV
+    format, respectively.
+
+    Args:
+        log_dir: Directory for the experiment logs
+    """
+
+    NAME_HPARAMS_FILE = 'hparams.yaml'
+    NAME_METRICS_FILE = 'metrics.csv'
+
+    def __init__(self, log_dir: str) -> None:
+        self.hparams = {}
+        self.metrics = []
+        self.metrics_keys = ["step"]
+
+        self.log_dir = log_dir
+        if os.path.exists(self.log_dir):
+            rank_zero_warn(
+                f"Experiment logs directory {self.log_dir} exists and is not empty. "
+                "Previous log files in this directory will be deleted when the new ones are saved!"
+            )
+        os.makedirs(self.log_dir, exist_ok=True)
+
+        self.metrics_file_path = os.path.join(self.log_dir, self.NAME_METRICS_FILE)
+
+    def log_hparams(self, params: Dict[str, Any]) -> None:
+        """Record hparams"""
+        self.hparams.update(params)
+
+    def log_metrics(self, metrics_dict: Dict[str, float], step: Optional[int] = None) -> None:
+        """Record metrics"""
+        def _handle_value(value):
+            if isinstance(value, torch.Tensor):
+                return value.item()
+            return value
+
+        if step is None:
+            step = len(self.metrics)
+
+        new_row = dict.fromkeys(self.metrics_keys)
```

Comment:
I do not think that is needed as it maintains only all future rows are complete but some may be missing in past...
I would rather just save as it is and do write a padding method for all rows according to the max